sszgen/hash.go

```diff
@@ -95,78 +95,112 @@ func (v *Value) hashRoots(isList bool, elem Type) string {
 	})
 }
 
-func (v *Value) hashTreeRoot() string {
+// takes a "name" param so that the variable name can be replaced with a local name
+// ie within a for loop for a list, the we want to refer to "elem" w/o a receiver variable
+// when not specified, name will be set to "::." + v.name. In the final templating pass,
+// the output formatter replaces all instances of "::" with the receiver variable for the container.
+func (v *Value) hashTreeRoot(name string, appendBytes bool) string {
+	if name == "" {
+		name = "::." + v.name
+	}
 	switch v.t {
 	case TypeContainer, TypeReference:
 		return v.hashTreeRootContainer(false)
 
 	case TypeBytes:
-		// There are only fixed []byte
-		name := v.name
 		if v.c {
 			name += "[:]"
 		}
-
-		tmpl := `{{.validate}}hh.PutBytes(::.{{.name}})`
-		return execTmpl(tmpl, map[string]interface{}{
-			"validate": v.validate(),
-			"name":     name,
-			"size":     v.s,
-		})
+		if v.isFixed() {
```

Comment:
Take a look at how I've changed isFixed. In particular note that I added a boolean so that when the parser sees a `Vector(Byte, N)` it can specifically mark the resulting byte Value as fixed.